sql/create_indicators_table.sql

```diff
@@ -0,0 +1,316 @@
+			count (DISTINCT date_part('week', inspectiondate::date)) FILTER (
+				WHERE inspectiondate >= (CURRENT_DATE - interval '1' year)
+			) AS hpd_viol_distress
+		FROM hpd_violations
+		WHERE class = any('{B,C}') 
+		GROUP BY bbl
+	),
+	hpd_comp AS (
+		SELECT 
+			bbl,
+		    count(*) FILTER (WHERE receiveddate >= (CURRENT_DATE - interval '6' month)) AS hpd_comp__6mo,
+		    count(*) FILTER (WHERE receiveddate >= (CURRENT_DATE - interval '1' month)) AS hpd_comp__1mo,
+		    count(*) FILTER (WHERE receiveddate >= (CURRENT_DATE - interval '7' day)) AS hpd_comp__week,
+		    count(*) FILTER (
+		    	WHERE receiveddate >= (CURRENT_DATE - interval '6' month) AND type = any('{IMMEDIATE EMERGENCY,HAZARDOUS,EMERGENCY}')
+		    ) AS hpd_emergency_comp__6mo,
+		    count(*) FILTER (
+		    	WHERE receiveddate >= (CURRENT_DATE - interval '7' day) AND type = any('{IMMEDIATE EMERGENCY,HAZARDOUS,EMERGENCY}')
+		    ) AS hpd_emergency_comp__week,
+			array_agg(DISTINCT apartment) FILTER (
+				WHERE apartment NOT IN ('BLDG','BLDNG','BUILDI','NA') AND receiveddate >= (CURRENT_DATE - interval '6' month)
+			) AS hpd_comp_apts__6mo,
+			array_agg(DISTINCT apartment) FILTER (
+				WHERE apartment NOT IN ('BLDG','BLDNG','BUILDI','NA') AND receiveddate >= (CURRENT_DATE - interval '7' day)
+			) AS hpd_comp_apts__week,
+			count (DISTINCT date_part('week', receiveddate::date)) FILTER (
+				WHERE receiveddate >= (CURRENT_DATE - interval '1' year)
+			) AS hpd_comp_distress
+		FROM hpd_complaints_and_problems
+		GROUP BY bbl
+	),
+	dob_ecb_stacked AS (
+		SELECT
+			bbl,
+			issuedate,
+			(violationcategory ~* 'ACTIVE') AS is_active
+		FROM dob_violations 
+		WHERE issuedate >= (CURRENT_DATE - interval '6' month)
+			AND violationtypecode IS NOT NULL
+		UNION
+		SELECT
+			bbl,
+			issuedate,
+			(certificationstatus IS NULL 
+			OR certificationstatus in ('REINSPECTION SHOWS STILL IN VIOLATION', 
+					'CERTIFICATE PENDING', 'CERTIFICATE DISAPPROVED', 'NO COMPLIANCE RECORDED')
+			) AS is_active
+		FROM ecb_violations 
+		WHERE issuedate >= (CURRENT_DATE - interval '6' month)
+			AND severity IS NOT NULL
+	),
+	dob_ecb_viol AS (
+		SELECT
+			bbl,
+			count(*) FILTER (WHERE issuedate >= (CURRENT_DATE - interval '6' month)) AS dob_ecb_viol__6mo,
+			count(*) FILTER (WHERE issuedate >= (CURRENT_DATE - interval '1' month)) AS dob_ecb_viol__1mo,
+			count(*) FILTER (WHERE issuedate >= (CURRENT_DATE - interval '7' day)) AS dob_ecb_viol__week
+		FROM dob_ecb_stacked
+		GROUP BY bbl
+	),
+	dob_comp AS (
+		SELECT
+			bin,
+			count(*) FILTER (WHERE dateentered >= (CURRENT_DATE - interval '6' month)) AS dob_comp__6mo,
+			count(*) FILTER (WHERE dateentered >= (CURRENT_DATE - interval '1' month)) AS dob_comp__1mo,
+			count(*) FILTER (WHERE dateentered >= (CURRENT_DATE - interval '7' day)) AS dob_comp__week
+		FROM dob_complaints
+		WHERE GREATEST(dateentered, dispositiondate, inspectiondate) >= (CURRENT_DATE - interval '1' year)
+		GROUP BY bin
```

Comment:
as we talked about with the issue of the wow bin being just a random one of possibly many, we can instead join here to assign bbl to all records and group by bbl instead. Then we can just take one of the bins from the complaints and use that when generating the link to dob website. 

```
	SELECT
			bbl,
			max(bin) AS bin,
			count(*) FILTER (WHERE dateentered >= (CURRENT_DATE - interval '6' month)) AS dob_comp__6mo,
			count(*) FILTER (WHERE dateentered >= (CURRENT_DATE - interval '1' month)) AS dob_comp__1mo,
			count(*) FILTER (WHERE dateentered >= (CURRENT_DATE - interval '7' day)) AS dob_comp__week
		FROM dob_complaints
		LEFT JOIN pad_adr USING(bin)
		WHERE GREATEST(dateentered, dispositiondate, inspectiondate) >= (CURRENT_DATE - interval '1' year)
		GROUP BY bbl
```